File: api/src/main/java/de/hipp/pnp/api/fivee/interfaces/FiveECharacterService.java
package de.hipp.pnp.api.fivee.interfaces;

import de.hipp.pnp.api.fivee.abstracts.BaseCharacter;

import java.util.List;

public interface FiveECharacterService<T extends BaseCharacter> {

    public List<T> getAllCharacters(String userId);

    public T generate();
}

File: api/src/main/kotlin/de/hipp/pnp/api/fivee/interfaces/FiveECharacterService.kt
package de.hipp.pnp.api.fivee.interfaces

import de.hipp.pnp.api.fivee.abstracts.BaseCharacter

interface FiveECharacterService<T : BaseCharacter> {
    fun getAllCharacters(userId: String?): MutableList<T>

    fun generate(): T?
}
